tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+        resp.media = self.status()
+
+    def terminate(self):
+        self.wlogger.debug("Terminating web produce class")
+        for t in self.topics.values():
+            t.terminate = True
+            t.producer.flush()
+        return
+
+
+class StreamVerifierConsume(StreamVerifier):
+    def __init__(self, cfg):
+        self.cfg = cfg
+        self.wlogger = setup_logger(LOGGER_WEB_CONSUME)
+        super().__init__(cfg.brokers,
+                         cfg.topics_config,
+                         cfg.worker_threads,
+                         rate=cfg.msg_rate,
+                         total_messages=cfg.msg_total)
+
+    def on_get(self, req: falcon.Request, resp: falcon.Response):
+        """Handles GET requests"""
+        self.wlogger.debug("Processing consume get request")
+        resp.status = falcon.HTTP_200  # This is the default status
+        resp.content_type = falcon.MEDIA_JSON  # Default is JSON, so override
+        resp.media = self.status()
+
+    def on_post(self, req: falcon.Request, resp: falcon.Response):
+        self.wlogger.debug("Processing consume post request")
+        # Validate request
+        if req.content_type != falcon.MEDIA_JSON:
+            resp.status = falcon.HTTP_400
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {"errors": ["Invalid request media type"]}
+            return
+
+        # Check active consuming event
+        if self.consume_thread is not None and \
+                self.consume_thread.is_alive():
+            resp.status = falcon.HTTP_400
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {"errors": ["Active produce job not finished"]}
+            return
+
+        # update topic config
+        topics_cfg_keys = ["topic_prefix", "topic_count"]
+        forbidden = ["msg_rate", "msg_total"]
+        error_msgs = validate_keys(req.media, topics_cfg_keys, forbidden)
+        if len(error_msgs) > 0:
+            msg = f"Incoming request invalid: {', '.join(error_msgs)}\n" \
+                f"Valid keys are: {', '.join(topics_cfg_keys)}"
+            self.wlogger.error(msg)
+            resp.status = falcon.HTTP_400
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {"errors": msg}
+        else:
+            # Update topic configs
+            if 'topic_prefix' in req.media:
+                self.topics_config.topic_prefix = req.media['topic_prefix']
+            if 'topic_count' in req.media:
+                self.topics_config.topic_count = req.media['topic_count']
+            # start consumers
+            self.init_consumers()
+            self.consume(wait=False)
+
+            resp.status = falcon.HTTP_200
+            resp.content_type = falcon.MEDIA_TEXT
+            resp.text = "OK"
+
+    def terminate(self):
+        self.wlogger.debug("Terminating web consume class")
+        for t in self.topics.values():
+            t.terminate = True
```

Comment:
we are marking the state but not waiting, is that ok? AFAICT we are not waiting even in the main method.

Reply:
Thread will terminate on flag and waiting is not needed in this case. It is a bit hacky way, but works